Celery utest coverage

alexandraBara opened the pull request.

alexandraBara committed:
checking coverage details

alexandraBara committed:
utest for new miopen_lib functions

alexandraBara committed:
move miopen_lib utests

alexandraBara committed:
utest fix for miopen_lib

alexandraBara committed:
fin_builder utest extra coverage for miopen_lib functionality

alexandraBara committed:
added test_celery utest

alexandraBara committed:
added more to celery utest

alexandraBara committed:
fixed import

alexandraBara committed:
extra tuna_celery code for utest

alexandraBara committed:
pylint

alexandraBara committed:
added test_celery to utest call

alexandraBara committed:
added test_helper for extra codecov

alexandraBara committed:
new test_helper utest

alexandraBara committed:
more utests for fin_eval and celery

alexandraBara committed:
more fin_eval utests

alexandraBara committed:
more celery functions utest added

alexandraBara committed:
yapf + test_mituna_interface

alexandraBara committed:
yapf + test_mituna_interface

alexandraBara committed:
cleanup

alexandraBara committed:
fixed utest test_celery check

alexandraBara committed:
ignoring example/ from coverage

cderb commented on tests/test_celery.py:
```diff
@@ -0,0 +1,277 @@
+                           ['miopen_perf_compile'],
+                           'test_add_celery_compile_job',
+                           'miopenConvolutionAlgoGEMM')
+  #assert num_jobs
+  num_jobs = 4
```
comment is assert, action is set

alexandraBara replied:
leftover from testing, ill remove

cderb commented on tests/test_celery.py:
```diff
@@ -0,0 +1,277 @@
+  subp_list = launch_worker_per_node([machine], cmd, True)
+  sleep(2)
+  assert subp_list
+  assert miopen.cancel_consumer(f"test_{db_name}")
+  for subp in subp_list:
+    print(subp.pid)
+    subp.kill()
```
cancel_consumer would also kill the workers for the named queue. Perhaps this loop would confirm that the subp were terminated by this call.

alexandraBara replied:
i'll add

alexandraBara committed:
fixed logger formatting

alexandraBara committed:
utest update post branch pull

alexandraBara committed:
Tuning through Celery (#970)

* fixed clinfo path

* removed comment out

* added new arch queue

* fixed new arch enqueue function

* removed testing code

* moved function over from fin_builder

* find and replace fix

* fix actuator func update

* merge develop fix

* removed celery runs from groovy file

* fixed applic maybe

* pylint

* moving code around to make celery integration modular

* updates to remove circ dependency

* ctrl+c and queue drain

* removed db query for job and config, using SimpleDict instead

* pylint and added TUNA_REDIS_PORT

* some renaming

* added timer

* utest update for moved files

* accounted for file rename

* registered tasks manually after task file was moved to miopen folder

* undid code change in example

* README updated

* legacy code update

* fixed SQL query syntax

* cleanup attemps + mlir test failure undo

* commented out failing rocmlir test assert

* pylint fix

* tuning updates

* duplicate logger handler

* pulled session out of processing function call

* utest fix

* syntax fix

* fixed the timing

* added subproc to purge Q by name

* pylint

* ctrl+c queue drain handle

* pylint

* stopping active workers on ctrl+x

* added enqueue_only arg

* merge fix

* added cancel_consumer, shutdown_worker cmd line arg + some review fixes

* moved cancel consumer to end of tuning

* pylint

* changing JK file to work with celery, added parallel step for findcompile

* fixed JK file syntax

* added kill subp by PID to terminate celery workers

* checking CI for execute step fail

* moved CompileEnqueue job checking stage over from execute

* added stage to build DB

* running applic to add session

* sleep on execute for 30 sec to ensure enqueue

* testing with launching celery worker from groovy file

* parallel steps inside the same stage

* parallel stage edit

* added daemonize

* running enqueue in background with &

* pipe-ing output to file

* stdout, stderr from subp

* stdout, stderr from subp

* reading stoud

* added loop exit

* removed blocking read

* launching celery worker from groovy file

* added missing celery_log file

* fixed celery cmd

* removed detached mode

* starting worker before enqueue

* removed the background job for enqueue

* start/join on subp for celery workers

* running enq in the background

* removed subp.start

* calling subproc.run/join

* removed process wait barrier + groovy file update with kill C worker by pid

* groovy syntax fix

mypy pylint ignore

declaring pid var in groovy

trying to get pid

groovy fix

Bump idna from 2.8 to 3.7 (#965)

Bumps [idna](https://github.com/kjd/idna) from 2.8 to 3.7.
- [Release notes](https://github.com/kjd/idna/releases)
- [Changelog](https://github.com/kjd/idna/blob/master/HISTORY.rst)
- [Commits](https://github.com/kjd/idna/compare/v2.8...v3.7)

---
updated-dependencies:
- dependency-name: idna
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

launching celery for eval per GPU

fixed get num_gpu script

fix loop index + added --enqueue to find_eval

groovy range

range fix

testing with int cast

loop for i in iter

testing different loop

for loop instead of .each

removed for loop

var rename

removed loops

removed printout from go_fish

removed for loop

added NonCPS directive

updated perf_compile and perf_eval with celery runs

updated perf_compile and perf_eval with celery runs

each

replaced elem with it

removed all loops

* pipeline fix

* hardcoding numbers for golden entries

* undid perf compile/eval shrink

* fix groovy

* undid merge of perf runs in perfEval

* added enqueue_only cmd line arg

* updated update_golden call to write entries with version 1

* commented out solverA test that checks for gfx908 entries (we run a diff arch for perf)

* added GPU loop for findEval

* fixed groovy echo

* fix pid list for celery workers, function for stopping workers in enqueue, enqueue purges queue at start

* pass subprocess pointers to tuning setup and join tuning setup to spawned celery workers

* kill celery workers when call to worker generation is canceled

* extra loggin

* added asyncio

* fixed counter

* resetting job state for in flight jobs with no results

* state rest

* move purge_queue

* trying to update redis key prefix

* trying to update redis key prefix

* only shutdown celery workers from current session

* global_keyprefix

* unified prefix nme

* updated Q name, global_prefix + result fetch on not new jobs

* moved functions into utility files

* no-cache

* moved enqueue into mituna_interface

* moved everything into interfaces

* update baseimage in dockerfile

* moved prep_tuning and tune in interface

* fixins

* fixed enum operation

* cleanup

* fixed Operation check

* worker operation fix

* fixed operation compare

* extra logging

* import

* removed app.start

* removed app.start so we can set configurations

* updated Q name in manual worker launch in groovy file

* moved machine init in run function

* cache machine for celery task to avoid re-init, retrieve worker name in celery task, set gpu_id in celery task

* using task_id to differentiate btw different runs and results

* cleanup

* fixed groovy file celery q names and worker names as per code updates

* cache tuna worker objects for celery workers

* added subproc for async function

* filter eval op for querying gpu_id in celery task

* custom error class + cleanup

* yapf

* README updates

* fixed missing args + CTRL+C interrupt catch for async run

* fix for worker_launch

* updates in groovy to print celery worker output

* syntax fix

* added logfile printout for compile step

* utest updates to accomodate for functions moving around

* removed no-cache from docker builds

* yapf

* removed pid from logname

* test

* launching workers through tuna

* inspecting celery

* adding args.fin_steps to redis key prefix to differentiate btw find/perf/eval/compile results in consume

* merged parallel branch

* removed local prefix

* string filter adjustment

* added cleaup function for redis results

* fixed launch for compile worker

* fixed prefix filtering for redis keys

* fix prefix issue when no fin_steps present

* fixed sess_str to cancel worker

* added reply_to for task decorator

* prefix fix

* string literal fix

* updated celery worker name in groovy file

* launching worker locally in detach mode

* set correct celery env vars to launch worker manually

* added CELERY_BROKER env var

* setup env vars for rest of utils functions that work with celery

* undid groovy utils function name change

* explicitly setting TUNA_CELERY_BROKER in docker env

* added CELERY_BROKER env var for tuna launch

* separated log files for stages

* added CELERY_BROKER env var to findEval

* moved CELERY_BROKER url to inline

* added CELERY_BROKER inline export for each enqueue_call

* print proc_id

* fix for pid return

* escaping &

* replace double quotes to eascap $

* fixed stdout/stderr redirect

* 123

* 123

* replaced &=046

* reverted single quotes to double

* cleanup

* cleanup

* undid false removal of load_job

* sudo for kill -9

* enabling perfCompile/eval stages

* fixed log file for perf_eval

* fixed celery worker name for perfEval

* fixed gpu_id for eval stages

* fixed groovy syntax

* fixed groovy syntax

* checking whoami /etc/pwd

* enabling back utests

* utest + doc updates

* fix for fin_builder utest

* syntax fix

* fin_eval utest update

* moved fin_eval utest step to gpu node and split from coverage

* fin_eval utest update

* yapf

* checking jobs

* skipping codecov to test all other CI stages

* utest fin_eval fix

* added celery_app and celery_tasks to ignore in codecov

* yapf

* fin_eval utest fic

* fineval stage try catch for printing celery logfile

* checking prefix

* fixed log file in find eval

* fixed groovy syntax

* removed sudo from kill -9

* try catch for pid kill

* try catch block for kill cmd in perfEval

* undid update_golden calls in groovy file

* canceling consumer on early exit for no new jobs

* celery launches for pipelines

* add env for celery broker

* end docker container running enqueue

* fix

* remote workers will be killed by enqueue, so remove extra kill

* kernel clear search performance

* update gpu_id tracking

* end pipeline enqueue docker with sigkill to allow cleanup

* remove docker wait

* Revert "remove docker wait"

This reverts commit 92930cc39a2d66f4ab492cf2f65f1d0aff6aca8e.

* reset job_counter on sigint

* periodic search for new jobs

* rebuild enqueue process

* reduce logging

* test speedup for kernel query

* add missing import

* test alchemy alternative

* test query

* cleanup

* cleanup + reviews

* addressed reviews

---------

Co-authored-by: Alex Bara <alex.bara@amd.com>
Co-authored-by: Christopher Erb <Christopher.Erb@amd.com>

alexandraBara committed:
merged develop

alexandraBara committed:
Merge branch 'alex_celery_cleanup' into alex_coverage

alexandraBara committed:
fix merge in JK file

alexandraBara committed:
fix merge in JK file

alexandraBara committed:
merge fix

alexandraBara committed:
readded .coveragec file post merge

alexandraBara committed:
added check for shutting down celery workers
